### PR TITLE
MM-32785 CenterChannel: Collapse Replies (show only root posts)

### PR DESCRIPTION
**Pull request by reflog**
#### Summary

Set collapsedThreads param based on user/site settings for post actions

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-32785

### CODE REVIEW

**Review by calebroseland — COMMENTED** [review]
So I tried this with the latest webapp+server `MM-32655`, and loading a channel directly causes a stuck loader for me. 

After initial load, if I switch teams, or go to global threads, then back to a channel, then it loads and shows only root posts.🤩

File: src/actions/posts.ts [hunk 1]
```diff
@@ -683,13 +684,14 @@ export function flagPost(postId: string) {
...
-export function getPostThread(rootId: string, fetchThreads = true, collapsedThreads = false, collapsedThreadsExtended = false) {
+export function getPostThread(rootId: string, fetchThreads = true, collapsedThreadsExtended = false) {
     return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const collapsedThreadsEnabled = isCollapsedThreadsEnabled(getState());
         dispatch({type: PostTypes.GET_POST_THREAD_REQUEST});
 
         let posts;
         try {
-            posts = await Client4.getPostThread(rootId, fetchThreads, collapsedThreads, collapsedThreadsExtended);
+            posts = await Client4.getPostThread(rootId, fetchThreads, collapsedThreadsEnabled, collapsedThreadsExtended);
```
[the remark this answers is not in the document]
Comment (reflog) [reply src/actions/posts.ts:694]: I think you are right. fixed

File: src/actions/posts.ts [hunk 2]
```diff
@@ -712,12 +714,12 @@ export function getPostThread(rootId: string, fetchThreads = true, collapsedThre
     };
 }
 
-export function getPosts(channelId: string, page = 0, perPage = Posts.POST_CHUNK_SIZE, fetchThreads = true, collapsedThreads = false, collapsedThreadsExtended = false) {
+export function getPosts(channelId: string, page = 0, perPage = Posts.POST_CHUNK_SIZE, fetchThreads = true, collapsedThreadsExtended = false) {
```
[the remark this answers is not in the document]
Comment (reflog) [reply src/actions/posts.ts:717]: collapsedThreadsExtended should come from the webapp (though I don't know if we need a case of collapsedThreadsExtended=true for webapp)

**Comment by calebroseland** [comment]
> loading a channel directly causes a stuck loader for me

This consistently happens if the latest post in a channel is a reply. If the latest is a root post, it loads properly.

**Comment by reflog** [comment]
@calebroseland - the issue was server side and it's now merged to master. please re-review

File: src/actions/posts.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (calebroseland) [reply src/actions/posts.ts:716]: @reflog I think there could be some cases potentially—but yeah, I think there'd be a high potential for extraneous data (existing profiles, overlapping, etc.).

**Review by calebroseland — APPROVED** [review]
LGTM. I might request a couple of changes for MM-32786 (ThreadFooter part to this), but I'll create a separate PR for that if need be.

**Review by ashishbhate — APPROVED** [review]

**Pull request merged**